src/index.js

```diff
@@ -1,106 +1,297 @@
+ *    development: process.env.NODE_ENV === "development",
+ *    base: "/public",
+ * });
+ *
+ * // load information from `eik.json` and the Eik server
+ * await eik.load();
+ *
+ * // when development is true script.value will be /public/script.js.
+ * // when development is false script.value will be
+ * // https://{server}/pkg/{name}/{version}/script.js
+ * // where {server}, {name} and {version} are read from eik.json
+ * const script = eik.file("/script.js");
+ *
+ * app.get("/", (req, reply) => {
+ *    reply.type("text/html; charset=utf-8");
+ *    reply.send(`<html><body>
+ *      <script
+ *        src="${script.value}"
+ *        ${script.integrity ? `integrity="${script.integrity}"` : ""}
+ *        type="module"></script>
+ *    </body></html>`);
+ * });
+ *
+ * app.listen({
+ *    port: 3000,
+ * });
+ *
+ * console.log("Listening on http://localhost:3000");
+ * ```
+ */
+export default class Eik {
+	#development;
+	#loadMaps;
+	#config;
+	#path;
+	#base;
+	#maps;
+
+	/**
+	 * @param {Options} options
+	 */
+	constructor({
+		development = false,
+		loadMaps = false,
+		base = "",
+		path = process.cwd(),
+	} = {}) {
+		this.#development = development;
+		this.#loadMaps = loadMaps;
+		this.#config = {};
+		this.#path = path;
+		this.#base = trimSlash(base);
+		this.#maps = [];
+	}
+
+	/**
+	 * Reads the Eik config from disk into the object instance, used for building {@link file} links in production.
+	 *
+	 * If {@link Options.loadMaps} is `true` the import maps
+	 * defined in the Eik config will be fetched from the Eik server for
+	 * use in {@link maps}.
+	 */
+	async load() {
+		this.#config = await helpers.getDefaults(this.#path);
+		if (this.#loadMaps) {
+			this.#maps = await fetchImportMaps(this.#config.map);
+		}
+	}
+
+	/**
+	 * The `"name"` field from the Eik config
+	 * @throws if read before calling {@link load}
+	 */
+	get name() {
+		if (this.#config.name) return this.#config.name;
+		throw new Error("Eik config was not loaded before calling .name");
+	}
+
+	/**
+	 * The `"version"` field from the Eik config
+	 * @throws if read before calling {@link load}
+	 */
+	get version() {
+		if (this.#config.version) return this.#config.version;
+		throw new Error("Eik config was not loaded before calling .version");
+	}
+
+	/**
+	 * The `"type"` field from the Eik config mapped to its URL equivalent (eg. "package" is "pkg").
+	 * @throws if read before calling {@link load}
+	 */
+	get type() {
+		if (this.#config.type && this.#config.type === "package") return "pkg";
+		if (this.#config.type) return this.#config.type;
+		throw new Error("Eik config was not loaded before calling .type");
+	}
+
+	/**
+	 * The `"server"` field from the Eik config
+	 * @throws if read before calling {@link load}
+	 */
+	get server() {
+		if (this.#config.server) return this.#config.server;
+		throw new Error("Eik config was not loaded before calling .server");
+	}
+
+	/**
+	 * The pathname to the base on Eik (ex. /pkg/my-app/1.0.0/)
```

Comment:
"to the base on Eik" does that mean the same as "to the base path in the current sink"?

Reply:
No, the base path of the sink would be closer to `/`. I mean for example `/pkg/my-app/1.0.0/`.

Reply:
riiight, so it's the base path for the current package.